include/lsst/afw/geom/CoordinateBase.h

```diff
@@ -55,6 +55,12 @@ class CoordinateBase {
     static int const dimensions = N;
     typedef Eigen::Matrix<T, N, 1, Eigen::DontAlign> EigenVector;
 
+    CoordinateBase(CoordinateBase const&) = default;
+    CoordinateBase(CoordinateBase&&) = default;
+    CoordinateBase& operator=(CoordinateBase const&) = default;
+    CoordinateBase& operator=(CoordinateBase&&) = default;
+    virtual ~CoordinateBase() = default;
```

Comment:
That seems like a strange thing to guarantee. Why bother using inheritance, then? Keeping it `virtual` seems safer.

Reply:
Sorry, I misspoke; we never *delete* its subclasses via a base class reference or pointer, which is much easier to guarantee.  In this case, we also very rarely use the classes via a base class reference or pointer, but that's not an essential part of the pattern.  For example, Eigen uses CRTP extensively, and the templated base classes are very much a part of the interface.  It also doesn't define its destructors as virtual, because the base classes are by definition only used in contexts in which you can safely `static_cast` them a more derived class that can be inferred from their template arguments.

It's probably best to just say that because CRTP is essentially compile-time polymorphism, many of the usual rules of C++ inheritance that are derived from *runtime* polymorphism don't apply.  I think a good case could be made that the implementations of `Point` and `Extent` don't actually benefit from these base classes much (they add a lot of boilerplate in exchange for reducing a bit of other code duplication), and I'd probably be supportive of a ticket to remove them, but we should absolutely not add a vtable to any of them.

Reply:
I have no idea what you mean by "rules of C++ inheritance"; the principles of inheritance are language-independent. But it is true that none of the classes in question contain other `virtual` methods, so they can't really be used as interfaces anyway. I'll change the declarations.